### PR TITLE
Support webhook notification

### PR DESCRIPTION
**Pull request by eisuke**
@cookpad/dev-infra 👓

### CODE REVIEW

File: lib/autoload/kuroko2/workflow/notifier/webhook.rb [hunk 1]
```diff
@@ -0,0 +1,172 @@
...
+        end
+
+        def notify_launch
+          if @definition.hipchat_notify_finished?
+            request(
+              build_payload(
+                action: 'notify_launch',
+                level: 'INFO',
+                subject: message_builder.launched_text,
+                message: @instance.logs.select{ |log| log.level == 'INFO' }.last.try!(:message),
```
Comment (adorechic) [inline lib/autoload/kuroko2/workflow/notifier/webhook.rb]: [nits][IMO] How about `logs.reverse.detect { ... }.try!` ?

File: lib/autoload/kuroko2/workflow/notifier/webhook.rb [hunk 2]
```diff
@@ -0,0 +1,172 @@
...
+        end
+
+        def notify_retrying
+          if @definition.hipchat_notify_finished?
+            request(
+              build_payload(
+                action: 'notify_retrying',
+                level: 'INFO',
+                subject: message_builder.retrying_text,
+                message: @instance.logs.last(2).first.message,
```
Comment (adorechic) [inline lib/autoload/kuroko2/workflow/notifier/webhook.rb]: Why don't you use `logs[-2]` ?

File: lib/autoload/kuroko2/workflow/notifier/webhook.rb (hunk 1 above)
Comment (eisuke) [inline lib/autoload/kuroko2/workflow/notifier/webhook.rb]: Refactored 87fe25d

File: lib/autoload/kuroko2/workflow/notifier/webhook.rb (hunk 2 above)
Comment (eisuke) [inline lib/autoload/kuroko2/workflow/notifier/webhook.rb]: The `logs` is a `ActiveRecord::Associations::CollectionProxy`.
IF `logs` size is 1, `logs.last(2).first` returns results. But `logs[-2]` returns `nil`.
I want to keep this `logs.last(2).first` behavior.

Comment (adorechic) [inline lib/autoload/kuroko2/workflow/notifier/webhook.rb]: ic

**Pull request merged**